Developer-Toolbox/Controllers/WeeklyChallengesController.cs

```diff
@@ -0,0 +1,322 @@
+            var weeklyChallenge = db.WeeklyChallenges
+                .Include(wc => wc.WeeklyChallengeExercises)
+                .ThenInclude(wce => wce.Exercise)
+                .FirstOrDefault(wc => wc.Id == id);
+
+            if (weeklyChallenge == null)
+            {
+                TempData["message"] = "The requested Weekly Challenge does not exist.";
+                TempData["messageType"] = "alert-danger";
+                return RedirectToAction("Index");
+            }
+
+            // Lista de ExerciseIds asociate acestui WeeklyChallenge
+            var exerciseIds = weeklyChallenge.WeeklyChallengeExercises
+                .Select(wce => wce.ExerciseId)
+                .ToList();
+
+            // Obținem utilizatorul curent
+            var userId = _userManager.GetUserId(User);
+
+            // Numărul total de exerciții asociate provocării
+            var nrTotal = exerciseIds.Count;
+
+            // Filtrăm soluțiile pentru a le lua doar pe cele care au data CreatedAt între StartDate și EndDate
+            var nrSolutii = db.Solutions
+                .Where(s => s.UserId == userId
+                    && s.Score == 100
+                    && s.ExerciseId.HasValue
+                    && exerciseIds.Contains(s.ExerciseId.Value)
+                    && s.CreatedAt >= weeklyChallenge.StartDate.Date // Folosim doar data (fără ora)
+                    && s.CreatedAt <= weeklyChallenge.EndDate.AddDays(1).Date) // Folosim doar data (fără ora)
+                .Select(s => s.ExerciseId) // Obținem doar ExerciseId-urile pentru soluții corecte
+                .Distinct() // Ne asigurăm că fiecare exercițiu este numărat o singură dată
+                .Count();
+
+            // Stocăm raportul în ViewBag
+            ViewBag.SolutionsSummary = $"{nrSolutii}/{nrTotal}";
+
+            // Transmitem WeeklyChallenge către view
+            ViewBag.WeeklyChallenge = weeklyChallenge;
+            return View();
+        }
+
+
+
+
+        // GET: Edit Weekly Challenge
+        [Authorize(Roles = "Admin")]
+        public IActionResult Edit(int id)
+        {
+            SetAccessRights();
+
+            // Preluăm WeeklyChallenge din baza de date, inclusiv exercițiile asociate
+            var weeklyChallenge = db.WeeklyChallenges
+                .Include(wc => wc.WeeklyChallengeExercises)
+                .ThenInclude(wce => wce.Exercise)
+                .FirstOrDefault(wc => wc.Id == id);
+
+            // Verificăm dacă WeeklyChallenge există
+            if (weeklyChallenge == null)
+            {
+                TempData["message"] = "The requested Weekly Challenge does not exist.";
+                TempData["messageType"] = "alert-danger";
+                return RedirectToAction("Index");
+            }
+
+            // Adăugăm lista de exerciții în ViewBag pentru a o folosi în formular
+            ViewBag.Exercises = db.Exercises.ToList();
```

Comment:
The exercises previously chosen should also be selected in the view.

Reply:
Thank you for the feedback! I've updated the code so that the previously chosen exercises are now selected in the view as well. The necessary adjustments have been made to ensure this functionality. Let me know if there's anything else you'd like to discuss!